R/viz.R

```diff
@@ -2261,47 +2182,513 @@ splendidHeatmap = function(this_matrix,
+    B.zero.coords = pull(unique(coding.maf[which(coding.maf$time_point == 2 & VAF == 0), "coord"]))
+    B.zero = dplyr::filter(coding.maf, coord %in% B.zero.coords)
+
+    C.rows = which(coding.maf$time_point == 3)
+    C.zero.coords = pull(unique(coding.maf[which(coding.maf$time_point == 3 & VAF == 0), "coord"]))
+    C.zero = dplyr::filter(coding.maf, coord %in% C.zero.coords)
+
+    coding.maf$category = "trunk"
+    coding.maf[which(coord %in% A.zero.coords), "category"] = "not-A"
+    coding.maf[which(coord %in% B.zero.coords), "category"] = "not-B"
+    coding.maf[which(coord %in% C.zero.coords), "category"] = "not-C"
+
+    just_gained_lg_all = dplyr::filter(coding.maf, Hugo_Symbol %in% lg & category != "trunk" )
+    just_gained_lg = dplyr::filter(coding.maf, Hugo_Symbol %in% lg & category != "trunk" & time_point == 3) %>%
+      mutate(time_point = time_point + 0.4)
+
+    just_trunk = dplyr::filter(coding.maf, Hugo_Symbol %in% lg & category == "trunk" & time_point == 1) %>%
+      mutate(time_point = time_point - 0.4)
+
+    ggplot(coding.maf, aes(x = time_point, y = VAF, group = coord, colour = category)) +
+      geom_point() +
+      geom_line(alpha = 0.3) +
+      geom_text_repel(data = just_gained_lg, aes(label = Hugo_Symbol), size = 4, segment.linetype = 0) +
+      geom_text_repel(data = just_trunk, aes(label = Hugo_Symbol), size = 4, segment.linetype = 0) +
+      ggtitle(title) +
+      theme_minimal()
+    return(TRUE)
+  }
+  return(FALSE)
+}
+
+#' Generate a plot for structural variant sub type distribution sorted per chromosome
+#'
+#' @param this_sample Sample to be plotted.
+#' @param plot_subtitle Subtitle for created plot.
+#' @param chr_select vector of chromosomes to be included in plot, defaults to autosomes.
+#' @param coding_only Optional. Set to TRUE to restrict to plotting only coding mutations.
+#' @param from_flatfile If set to true the function will use flat files instead of the database.
+#' @param use_augmented Boolean statement if to use augmented maf, default is FALSE.
+#'
+#' @return Nothing.
+#' @export
+#'
+#' @examples
+#' sv_plot = fancy_sv_chrdistplot(this_sample = "HTMCP-01-06-00422-01A-01D")
+#' 
+fancy_sv_chrdistplot = function(this_sample,
+                                plot_subtitle = "Structural Variant Distribution Per Chromosome",
+                                chr_select = paste0("chr", c(1:22)),
+                                coding_only = FALSE,
+                                from_flatfile = FALSE,
+                                use_augmented_maf = FALSE){
+  
+  #get maf data for a specific sample.  
+  maf = assign_cn_to_ssm(this_sample = this_sample, coding_only = coding_only, from_flatfile = from_flatfile, use_augmented_maf = use_augmented_maf)$maf
+  
+  #convert variables to factors
+  maf$Variant_Type = as.factor(maf$Variant_Type)
+  maf$Chromosome = as.factor(maf$Chromosome)
+  
+  #add chr prefix if missing
+  if(!str_detect(maf$Chromosome, "chr")[5]){
```

Comment:
Curious why #5? Is it to check some element? You can probably use dplyr here to mutate the column. Although I think it is safe that assign_cn_to_ssm always returns consistent formatting since prefixing is handled in the database or in the flatfile.